app/models/ems_azure.rb

```diff
@@ -29,17 +30,16 @@ def connect(options = {})
     self.class.raw_connect(clientid, clientkey, tenant_id)
   end
 
-  def verify_credentials(_auth_type = nil, _options = {})
-    # TODO
-    true
-  end
-
-  def tenant_id=(tenant_id)
-    self.uid_ems = tenant_id
-    save
-  end
+  def verify_credentials(*)
+    begin
+      connect
```

Comment:
But I don't think that's how it's supposed to work...verify can be called with alternate credentials via the options, and those are the ones that would need to be verified (I think)